### DISCUSSION TITLE
How do I embed iframe into quarto page with full width and height?

### DISCUSSION THREAD

**Discussion by ollyhensby** [quarto-dev/quarto-cli#10498] [Q&A] [answered]
### Description

````qmd
---
pagetitle: "example"
---

::: {.column-screen}
<iframe src="https://www.openstreetmap.org/export/embed.html?bbox=-0.004017949104309083%2C51.47612752641776%2C0.00030577182769775396%2C51.478569861898606&layer=mapnik" style="width:100%; height:100%;">
</iframe>
:::
````

Which comes out with its height squished:

![image](https://github.com/user-attachments/assets/c9a02aab-38b6-4029-a59b-9c871039b8c1)

Is there a way to inherit the height from the website being referenced? Or maybe there is a better way to embed an iframe on a quarto page to take up the full width and height of the page?

**Comment by mcanouil** [comment, marked as answer]
A website does not usually have height.
You need a container with height set.

Note that this is an HTML/CSS question rather than a Quarto question as Quarto HTML formats still obeys HTML/CSS rules.

**Reply by ollyhensby** [reply]
Ok! Thank you for answering so quickly.

**Comment by cderv** [comment]
For anyone hitting this: there's a community extension, [quarto-iframe](https://github.com/beck-chan/quarto-iframe) by @beck-chan, that wraps an `<iframe>` in a fenced div with width/height/zoom/alignment options and an optional caption. Externally contributed extension, not maintained by the Quarto project, not tried by me — but it *claims* to solve exactly this sizing problem.

**Reply by mcanouil** [reply]
There is the older (still maintained): [embedio](https://github.com/coatless-quarto/embedio) by @coatless.

Note also that Pandoc (thus Quarto) can do native embedding:

````qmd
---
title: Embed
format: html
---

![](https://example.com/index.html){title="Example site" width="90%" height=720 align="left"}
````

**Reply by cderv** [reply]
Thanks for additional resources and example !
